src/zenml/integrations/kubernetes/metadata_stores/kubernetes_metadata_store.py

```diff
@@ -0,0 +1,376 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of Kubernetes metadata store."""
+
+import os
+import subprocess
+import time
+from typing import Any, ClassVar, Dict, Union
+
+from kubernetes import client as k8s_client
+from ml_metadata.proto import metadata_store_pb2
+from pydantic import root_validator
+
+from zenml.exceptions import ProvisioningError, StackComponentInterfaceError
+from zenml.integrations.kubernetes import KUBERNETES_METADATA_STORE_FLAVOR
+from zenml.integrations.kubernetes.orchestrators import kube_utils
+from zenml.io import fileio
+from zenml.logger import get_logger
+from zenml.metadata_stores.mysql_metadata_store import MySQLMetadataStore
+from zenml.utils import io_utils, networking_utils
+
+logger = get_logger(__name__)
+
+DEFAULT_KUBERNETES_METADATA_LOCAL_HOST_NAME = "127.0.0.1"
+DEFAULT_KUBERNETES_METADATA_DAEMON_TIMEOUT = 60
+DEFAULT_KUBERNETES_METADATA_DAEMON_PID_FILE = "kubernetes_metadata_daemon.pid"
+DEFAULT_KUBERNETES_METADATA_DAEMON_LOG_FILE = "kubernetes_metadata_daemon.log"
+
+
+class KubernetesMetadataStore(MySQLMetadataStore):
+    """Kubernetes metadata store (MySQL database deployed in the cluster).
+
+    Attributes:
+        deployment_name: Name of the kubernetes deployment and corresponding
+            service/pod that will be created when calling `provision()`.
+        kubernetes_context: Name of the kubernetes context in which to deploy
+            and provision the MySQL database.
+        kubernetes_namespace: Name of the kubernetes namespace.
+            Defaults to "default".
+        storage_capacity: Storage capacity of the metadata store.
+            Defaults to `"10Gi"` (=10GB).
+    """
+
+    deployment_name: str
+    kubernetes_context: str
+    kubernetes_namespace: str = "zenml"
+    storage_capacity: str = "10Gi"
+    _local_host_name: str = DEFAULT_KUBERNETES_METADATA_LOCAL_HOST_NAME
```

Comment:
Why the need to use a private attribute assigned to a constant ? Just use the constant directly.